### PR TITLE
Format contract id in fn_call event topic

### PR DESCRIPTION
**Pull request by sisuresh**
### What

Resolves https://github.com/stellar/rs-soroban-env/issues/1463

Before 
```
Event log (newest first):
0: [Diagnostic Event] contract:CAXDPD4A5ANAASQ2U74RBW7ULGB3TS5SGSAFYPN6O2R2IJKGEX22SWUS, topics:[error, Error(Contract, https://github.com/stellar/rs-soroban-env/pull/8)], data:["negative amount is not allowed", -1]
1: [Diagnostic Event] topics:[fn_call, Bytes(2e378f80e81a004a1aa7f910dbf45983b9cbb234805c3dbe76a3a4254625f5a9), mint], data:[GD4YHMIZ7IGW7E4XJ4TXIJPLJU255JXPM2PKTV6H4PA4N5RTM6T2R73D, -1]
2: [Diagnostic Event] contract:CAXDPD4A5ANAASQ2U74RBW7ULGB3TS5SGSAFYPN6O2R2IJKGEX22SWUS, topics:[fn_return, init_asset], data:Void
3: [Diagnostic Event] topics:[fn_call, Bytes(2e378f80e81a004a1aa7f910dbf45983b9cbb234805c3dbe76a3a4254625f5a9), init_asset], data:Bytes(00000001616161610000000000fc065517dbbca3e37ec0e97dbfcd80e6bcdfdac732be1e4152b259f1f7ea88)
```

After
```
Event log (newest first):
0: [Diagnostic Event] contract:CAXDPD4A5ANAASQ2U74RBW7ULGB3TS5SGSAFYPN6O2R2IJKGEX22SWUS, topics:[error, Error(Contract, https://github.com/stellar/rs-soroban-env/pull/8)], data:["negative amount is not allowed", -1]
1: [Diagnostic Event] topics:[fn_call, CAXDPD4A5ANAASQ2U74RBW7ULGB3TS5SGSAFYPN6O2R2IJKGEX22SWUS, mint], data:[GD4YHMIZ7IGW7E4XJ4TXIJPLJU255JXPM2PKTV6H4PA4N5RTM6T2R73D, -1]
2: [Diagnostic Event] contract:CAXDPD4A5ANAASQ2U74RBW7ULGB3TS5SGSAFYPN6O2R2IJKGEX22SWUS, topics:[fn_return, init_asset], data:Void
3: [Diagnostic Event] topics:[fn_call, CAXDPD4A5ANAASQ2U74RBW7ULGB3TS5SGSAFYPN6O2R2IJKGEX22SWUS, init_asset], data:Bytes(00000001616161610000000000fc065517dbbca3e37ec0e97dbfcd80e6bcdfdac732be1e4152b259f1f7ea88)
```

### CODE REVIEW

File: soroban-env-host/src/events/mod.rs [hunk 1]
```diff
@@ -123,11 +123,33 @@ impl core::fmt::Display for HostEvent {
...
                         write!(f, ", ")?;
                     }
-                    display_scval(topic, f)?;
+
+                    // The second topic of the fn_call event is the contract id as ScBytes,
+                    // but we want to display it as a C key instead, so this block
+                    // tries to deduce if the event is the fn_call event.
+                    if i == 1 && is_fn_call {
+                        if let ScVal::Bytes(bytes) = topic {
+                            let hash: [u8; 32] = bytes.clone().0.try_into().unwrap();
```
Comment (sisuresh) [inline soroban-env-host/src/events/mod.rs:138]: I'll remove this unwrap

File: soroban-env-host/src/events/mod.rs [hunk 2]
```diff
@@ -123,11 +123,33 @@ impl core::fmt::Display for HostEvent {
...
                 for (i, topic) in ceb.topics.iter().enumerate() {
                     if i != 0 {
                         write!(f, ", ")?;
                     }
-                    display_scval(topic, f)?;
+
+                    // The second topic of the fn_call event is the contract id as ScBytes,
+                    // but we want to display it as a C key instead, so this block
+                    // tries to deduce if the event is the fn_call event.
+                    if i == 1 && is_fn_call {
```
Comment (sisuresh) [inline soroban-env-host/src/events/mod.rs:136]: This check isn't perfect, but I don't think we need it to be.

Comment (dmkozh) [reply soroban-env-host/src/events/mod.rs:136]: Yeah, I think this is good enough, given that we only format events in debug setting and the ability to imitate fn_call diagnostic doesn't seem problematic for this case.

File: soroban-env-host/src/events/mod.rs (hunk 1 above)
Comment (dmkozh) [reply soroban-env-host/src/events/mod.rs:138]: This should actually be handled gracefully on the off-chance if event isn't actually a real fn_call event. We should check the size first and just fallback to the default option if it's not 32.

File: soroban-env-host/src/events/mod.rs [hunk 3]
```diff
@@ -123,11 +123,33 @@ impl core::fmt::Display for HostEvent {
...
+                            let strkey = stellar_strkey::Contract(hash);
+                            write!(f, "{}", strkey)?
+                        } else {
+                            display_scval(topic, f)?;
+                        }
+                    } else {
+                        display_scval(topic, f)?;
+                    }
+
+                    if let ScVal::Symbol(first_topic_str) = topic {
```
Comment (dmkozh) [inline soroban-env-host/src/events/mod.rs:148]: This should also be have a condition `if i==0`, right?

File: soroban-env-host/src/events/mod.rs (hunk 1 above)
Comment (sisuresh) [reply soroban-env-host/src/events/mod.rs]: Yup that's what I meant by removing the unwrap. Fixed

**Review by dmkozh — APPROVED** [review]

**Pull request merged**